sway-core/src/language/ty/expression/expression_variant.rs

```diff
@@ -614,7 +617,46 @@ impl ReplaceDecls for TyExpressionVariant {
                 ref mut arguments,
                 ..
             } => {
-                function_decl_id.replace_decls(decl_mapping, type_engine);
+                // Filter decl mapping so function declaration is replaced by another
+                // function declaration that matches argument types
+                // This is needed for when multiple trait constraints are used that replace the same trait function
```

Comment:
Can you elaborate on this? Maybe a code example? Thanks!

Reply:
This is required when we have a function such as:
```rust
fn function<T, E>(v1: T, v2: E) where T: Eq, E: Eq {
  v1 == v1
  v2 == v2
}
```
if T and E are different types then `v1 == v1` should call one trait implementation while `v2 == v2` should call another one. In this case decl_mapping will have `v1 == v1` function declarations replaced by the T implementation and  `v2 == v2` function declarations replaced by the E implementation. The code below filters the correct implementation to use.